src/cli/services/linter.ts

```diff
@@ -0,0 +1,128 @@
+import { isAbsolute, resolve } from '@stoplight/path';
+import { IParserResult, Optional } from '@stoplight/types';
+import { getLocationForJsonPath, parseWithPointers } from '@stoplight/yaml';
+
+import { readParsable } from '../../fs/reader';
+import { httpAndFileResolver } from '../../resolvers/http-and-file';
+import { getDefaultRulesetFile } from '../../rulesets/loader';
+import { isOpenApiv2, isOpenApiv3 } from '../../rulesets/lookups';
+import { readRuleset } from '../../rulesets/reader';
+import { Spectral } from '../../spectral';
+import { IParsedResult, RuleCollection } from '../../types';
+import { ILintConfig } from '../../types/config';
+import { IRuleset } from '../../types/ruleset';
+
+export async function loadRulesets(cwd: string, rulesetFiles: string | string[] | null): Promise<IRuleset> {
```

Comment:
Looking at this code it's clear that if `rulesetFiles` is empty or null — nothing will effectively happen. 

You can make this function type-safe (and avoid the if and returning empty stuff) by declaring that the array must not be empty

```suggestion
export async function loadRulesets(cwd: string, rulesetFiles: string[] & { 0: string }): Promise<IRuleset> {
```

Reply:
It's okay if it's empty, isn't it? I mean, if one wants to do so, why not.

Reply:
if `rulesetFiles: string[]`, then you only have to worry about one data structure though

Reply:
Yeah, it's mostly for other developers' convenience, since that function is accessible.
I can move that to `loadRuleset` method, don't have any preference, up to you.

Reply:
heh, don't know what moving "that to loadRuleset" solves. If it is about limiting number of data structures, then yeah, let's go for it :)